lib/Command/ExpirePassword.php

```diff
@@ -106,54 +129,128 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
+		$groups = $input->getOption('group');
+		$allUsers = $input->getOption('all');
+
+		if ($allUsers !== false) {
+			$users = [];
+			$function = function (IUser $user) use (&$users) {
+				$users[] = $user;
+			};
+
+			$this->userManager->callForAllUsers($function);
+			$this->setPasswordExpiryForUsers($input, $output, $users);
+		} elseif (\count($groups) >= 1) {
+			$users = [];
+
+			foreach ($groups as $group) {
+				if ($this->groupManager->groupExists($group) === false) {
+					$output->writeln("Group name $group doesn't exist");
+					return 1;
+				} else {
+					$users[$group] = [];
+					foreach ($this->getAllUsersFromGroup($group) as $users_array) {
+						$users[$group] = $users_array;
+						$this->setPasswordExpiryForUsers($input, $output, $users[$group]);
+					}
+				}
+			}
+		} else {
 
-		/** @var $user \OCP\IUser */
-		$user = $this->userManager->get($uid);
+			$uid = $input->getArgument('uid');
 
-		if ($user === null) {
-			$output->writeln("<error>Unknown user: $uid</error>");
-			return self::EX_NOUSER;
+			/** @var $user \OCP\IUser */
+			$user = $this->userManager->get($uid);
+
+			if ($user === null) {
+				$output->writeln("<error>Unknown user: $uid</error>");
+				return self::EX_NOUSER;
+			}
+
+			if (!$user->canChangePassword()) {
+				$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
+				return self::EX_GENERAL_ERROR;
+			}
+
+			if ($user !== null) {
+				$this->setPasswordExpiryForUsers($input, $output, [$user]);
+			}
 		}
 
-		if (!$user->canChangePassword()) {
-			$output->writeln("<error>The user's backend doesn't support password changes. The password cannot be expired for user: $uid</error>");
-			return self::EX_GENERAL_ERROR;
+		return 0;
+	}
+
+	protected function getAllUsersFromGroup($group) {
+		$count = 0;
+		$users = [];
+		foreach ($this->groupManager->findUsersInGroup($group) as $user) {
+			\array_push($users, $user);
+			$count++;
+			//Take 200 users at a time
+			if ($count > 199) {
+				yield $users;
+				$count = 0;
+				$users = [];
+			}
+		}
+		if (\count($users) > 0) {
+			yield $users;
 		}
+	}
 
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @param IUser[] $users
+	 */
+	private function setPasswordExpiryForUsers(InputInterface $input, OutputInterface $output, $users) {
 		$expireDate = new \DateTime();
-		$expireDate->setTimezone(new \DateTimeZone('UTC'));
+		$oldDate = new \DateTime();
+		$timeZone = new \DateTimeZone('UTC');
+		$inputExpireDate = $input->getArgument('expiredate');
+		foreach ($users as $user) {
+			$calculatedExpireDate = $this->setPasswordExpiry($expireDate, $oldDate, $timeZone, $inputExpireDate, $user);
+			if ($input->hasArgument('expiredate')) {
```

Comment:
Better add a comment here. It's weird that you get an argument, use it and later check if the argument is present.
I guess this `hasArgument` checks if the argument has been explicitly provided, while the `getArgument` might return the default value. I'm not sure if we should just write the message regardless of this...

Reply:
Removed the usage of checking it. In the latest version of the PR, its done in the `execute`.